WMF Framework/Event Platform/EventPlatformClient.swift

```diff
@@ -183,39 +184,22 @@ public class EventPlatformClient: NSObject, SamplingControllerDelegate {
     }
     private var _sessionID: String?
 
-    /**
-     * Store events until the library is finished initializing
-     *
-     * The EPC library makes an HTTP request to a remote stream configuration
-     * service for information about how to evaluate incoming event data. Until
-     * this initialization is complete, we store any incoming events in this
-     * buffer.
-     *
-     * Only modify (append events to, remove events from) *synchronously* via
-     * `queue.sync`
-     */
-    private var inputBuffer: [EventRequest] = []
-
-    /**
-     * Maximum number of events allowed in the input buffer
-     */
-    private let inbutBufferLimit = 128
-
-    /**
-     * Holds events that have been scheduled for POSTing
-     */
-    private var outputBuffer: [EventRequest] = []
 
     // MARK: - Methods
 
     public override init() {
+        self.storageManager = StorageManager.shared
         self.samplingController = SamplingController()
         
         super.init()
 
         self.samplingController.delegate = self
 
-        self.fetchStreamConfiguration(retries: 10, retryDelay: 30)
+        if (self.storageManager != nil) {
+            self.fetchStreamConfiguration(retries: 10, retryDelay: 30)
+        } else {
+            DDLogError("EPC: Error initializing the storage manager. Event intake and submission will be disabled.")
+        }
```

Comment:
Done.  Actually, while we're talking about this section: @linehan and I had our weekly tech huddle a couple of hours ago and were thinking about whether the client should be disabled completely in the (unlikely?) event that the storage setup goes sideways, or whether it should continue to attempt to send events through without permanent storage like it's doing now.  Our tentative conclusion was that it would be best to simply fail, as this PR is currently written, in the first instance, and perhaps do some beta testing to see how prevalent storage setup failures are in the wild and whether it turns out that they are prevalent enough to justify adding the complexity of a fallback code path.  What do you think?

Reply:
@mdholloway sure that sounds good to me, that is, don't add the complexity unless it's prevalent enough. Do you have thoughts on how to measure storage setup failures? Is the plan to hit a logging endpoint if `self.storageManager = nil` to mark an instance of a storage failure, or are we going to assume `edit_history_compare` count differences from the legacy system mean the storage setup failed?

Reply:
I didn't have anything in mind beyond the latter.  Would it be possible to determine the effect (if any) based only on the numbers of events coming back from beta testers, or is that pool too small to tell us anything conclusive?

Reply:
@tonisevener There are a couple of paths that could lead the event system to disable itself. At least user opt-out, stream configuration fetch failure, and storage failure, though there may be more. I could imagine all of these going to the same method to disable the system, perhaps this method takes a `reason` argument, but at least stream configuration fetch and storage failure could generate an event and POST it directly to one of our backends (similar to how client error logging is done in MediaWiki). If a POST couldn't happen immediately (we'd expect this if stream configuration fetch failure occurred), perhaps this diagnostic event could be persisted until the next time the device is online.

Something like this was my thought. I don't think we have sufficient information to diagnose a delta between new and legacy as being caused by storage failure otherwise.

Reply:
@linehan your error logging all sounds great, should we maybe think about putting that in for a v3?

Reply:
@tonisevener Yeah, let's revisit for v3.